### ISSUE TITLE
Water is not rendered for parks extending over coastline

### ISSUE THREAD

**Issue by artischocke** [gravitystorm/openstreetmap-carto#2609]
In South Florida, I've come across cases where a park extending over coastline areas results in water not being rendered over the green color of the park. One such example is [Oleta River State Park](http://www.openstreetmap.org/way/119291279)
![1](https://cloud.githubusercontent.com/assets/19168045/24948133/9b150cd6-1f69-11e7-9a81-2a40cf333aec.PNG)

where, aside from the very odd location chosen by the rendered for the park's label, coastline features aren't visible over the green cover of the park. Here's the same view with the coastline features visible using OSMI's Coasline view
![2](https://cloud.githubusercontent.com/assets/19168045/24948232/f56084d6-1f69-11e7-871b-1d7a1f9dcbb5.PNG)

I had worked around this issue by creating relations which included the coastline way as the park's boundary, but in this case I believe it would be incorrect to exclude the surface of the water from the park's limits just to get the water to render (i.e. mapping for the renderer), so I thought I'd ask whether this was intended behavior or a bug. If the former, what would be a reasonable but still correct workaround?

**Comment by imagico** [comment]
Here specifically could be considered a tagging problem, i.e. you want to use `leisure=park` for something other than what it is meant for (open, green area for recreation)

More generally this is a duplicate of #1473 which is meant to be be fixed by #1982.

**Comment by imagico** [comment]
The label is indeed wrong by the way, does not appear to be a data glitch since it is the same in the German style - looks like a Mapnik bug to me - likely https://github.com/mapnik/mapnik/issues/3558

**Comment by artischocke** [comment]
@imagico thanks for the answer, but I would say "open, green area for recreation" is a very narrow definition of what a park can be. I take 'open' to mean outdoors as opposed to covered (otherwise places with thick tree cover wouldn't fit the definition), and 'green' would necessitate that recreation activities take place exclusively over grass. Even if we ignore the fact that the name includes the word _Park_ , it would seem to me that the site is indeed a park as most of its [attributes and facilities](https://www.floridastateparks.org/park/Oleta-River) match the definition offered by [Wikipedia](https://en.wikipedia.org/wiki/Park).

All that aside, the official park boundaries are over coastal waters, so whether we think that's correct or not, it's simply something that exists in the wild.

![1](https://cloud.githubusercontent.com/assets/19168045/24969623/a09d126e-1fb1-11e7-9596-5250940b9fb3.PNG)

So if I'm understanding correctly, the takeaway here is that a `leisure=park` cannot (due to technical limitations) extend over coastlines, correct? The park is alternatively called Oleta River _Recreational Area_. Would I encounter the same behavior if the park were tagged `landuse=recreation_ground`. I feel that's the wrong tag and tagging for the renderer, but I'm curious if it would yield the same result.

Unfortunately, there are far more complex cases in Florida, such as [West Lake Park](http://www.openstreetmap.org/way/123658169) just farther north. Here, there is a bay (which is also not rendered) with many many more features involved. The alternative of mapping this with the coastline ways on the intracoastal and all others in a `natural=water` relation seems unintuitive and unnecessarily complex. Thanks again for your feedback..

![2](https://cloud.githubusercontent.com/assets/19168045/24969991/0b9c5920-1fb3-11e7-975e-0e50256402ab.PNG)

**Comment by dieterdreist** [comment]
sent from a phone

> On 12. Apr 2017, at 19:06, artischocke <notifications@github.com> wrote:
> 
> Unfortunately, there are far more complex cases in Florida, such as West Lake Park just farther north. Here, there is a bay (which is also not rendered) with many many more features involved.


are you sure this should be mapped as leisure=park? Seems a good candidate for leisure=nature_reserve

a leisure=park is a man_made landscape, purposefully arranged "natural" objects

**Comment by imagico** [comment]
Ok, let's not start a tagging discussion here.  I just wanted to mention the possibility that excluding this water area from the park might actually be the correct way to map this.   This is not because of the way it is rendered here though - the current rendering behavior is not in any way meant to suggest that certain way of mapping things are correct and others are not.  The idea simply comes from the current meaning of the tag.

If you want to discuss where exactly to use `leisure=park` and how to tag official park boundaries for some meaning of the term 'park' better places would be the tagging mailing list or the OSM wiki.

**Comment by dieterdreist** [comment]
2017-04-12 23:12 GMT+02:00 Christoph Hormann <notifications@github.com>:

> Ok, let's not start a tagging discussion here. I just wanted to mention
> the possibility that excluding this water area from the park might actually
> be the correct way to map this. This is not because of the way it is
> rendered here though - the current rendering behavior is not in any way
> meant to suggest that certain way of mapping things are correct and others
> are not. The idea simply comes from the current meaning of the tag.
>
> If you want to discuss where exactly to use leisure=park and how to tag
> official park boundaries for some meaning of the term 'park' better places
> would be the tagging mailing list or the OSM wiki.
>


I believe tagging is important here. If you accept that parks are about
"man made nature", this might exclude coastal waters in general, so there'd
be no need to revisit the rendering for these. The "park" in question is
described like this on the web:

"The ecological importance of the relationships between mangroves, our
coastal waters, and the adjacent uplands cannot be overemphasized. West
Lake is the largest remaining mangrove ecosystem in the 85-mile, highly
developed urban coastal zone from Miami Beach to West Palm Beach. This
vital coastal land provides valuable natural resources and significant
recreational opportunities to urban residents and visitors."
from: http://www.broward.org/Parks/WestLakePark/Pages/Default.aspx

look at terms like "remaining", "natural resources", and "mangroves",
clearly a natural reserve (IMHO).

**Comment by artischocke** [comment]
@dieterdreist thanks for your thoughts regarding tagging. I would agree that West Lake fits the definition of a natural reserve better than that of a park. Unfortunately, the distinction isn't often clear to mappers, especially in cases where the name would seem to indicate they're parks. I'd maintain, however, that the in the case of the original question, Oleta River, is indeed a park. Secondly (and most relevant to this discussion), my point was that (correctly tagged) parks _might_ extend over coastlines, at least with respect to the official park boundaries. Whether the renderer chooses to apply an appropriate layering i such cases seems to me more a question of being strict about the semantics of a tag vs. being lenient & practical and accommodating real-world scenarios.

The only question that remains open for me is whether the same would happen with a recreation ground. Thanks again for your comments.

**Comment by dieterdreist** [comment]
sent from a phone

> On 15. Apr 2017, at 13:02, artischocke <notifications@github.com> wrote:
> 
> The only question that remains open for me is whether the same would happen with a recreation ground.


if you believe that "parks" which are (partly) in the sea are correctly tagged with leisure=park (which I am also not categorically refusing on second thought, e.g. there could be something belonging to a park in the waters nearby, hence extending the park into the sea), then the same might be true for recreation grounds as well (besides maybe the actual meaning of the word "ground")

**Comment by mboeringa** [comment]
> being lenient & practical and accommodating real-world scenarios.

This is the difference between a hand-drawn map and a fully automatically created one like this. There are technical hurdles that make accommodating every wish of of single user impossible. You are more or less asking for a hand drawn map with this request.

As to the technical side of this issue: there is a cartographic choice of the developers of this style to render parks with a solid light green fill, which probably goes back to historic maps and is also frequently used in other styles (e.g. Google). 

Solid-on-solid can't be displayed at the same time. Sea/ocean bodies are rendered **_below_** anything else, which is necessary as otherwise all islands and islets need to be cut from all waterbodies world wide. This would result in unmaintainable large MPs. Hence the park will cover the sea/ocean when you draw it across the coast.

A solution would be to render all parks as outline or even only with a label and no physical representation like an outline or fill. That requires the park to have a name to be visible (which is probably not a big issue, since most parks have them).

However, that breaks consistency with past choices. Such a potential change though, is neither good nor bad, just as the original choice to render with solid fill isn't good or bad. It is simply _**a cartographic styling choice**_, nothing else.

**Comment by artischocke** [comment]
@mboeringa thanks for your input. I understand there are technical reason why this doesn't work as I mentioned in my first comment to @imagico, and I'm satisfied with that answer so long as the limitation is well understood (it's clearly well understood in this community, but it was unexpected behavior to me as a mapper as I imagined water features being on the map as being important).

> You are more or less asking for a hand drawn map with this request.

I was simply asking whether this was intended behavior or a bug :-) I was also looking for your feedback in an appropriate workaround in case it was intended behavior. That was my intention when writing up this issue, just in case there was something which needed fixing, or otherwise find a way to make it work within reason given the limitations.

To recap, I'm fine with the rendering of parks, I understand why it happens and why it is a limitation of the renderer. That said, I don't subscribe to the view any park extending over coastal waters is incorrectly mapped de facto and that all parks should be restricted to land and "open green spaces"; there are be valid uses of parks which extend over coastal waters (same goes for recreation areas).

Thanks again for taking the time to get back to me on this so quickly comprehensively! Cheers!
